airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1053,18 +1054,29 @@ def create_concurrent_cursor_from_datetime_based_cursor(
             if evaluated_step:
                 step_length = parse_duration(evaluated_step)
 
-        clamping_strategy = NoClamping()
+        clamping_strategy: ClampingStrategy = NoClamping()
         if datetime_based_cursor_model.clamping:
-            match datetime_based_cursor_model.clamping.target:
-                case Clamping.target.DAY:
+            # While it is undesirable to interpolate within the model factory (as opposed to at runtime),
+            # it is still better than shifting interpolation low-code concept into the ConcurrentCursor runtime
+            # object which we want to keep agnostic of being low-code
+            target = InterpolatedString(
+                string=datetime_based_cursor_model.clamping.target,
+                parameters=datetime_based_cursor_model.parameters or {},
+            )
+            evaluated_target = target.eval(config=config)
+            match evaluated_target:
+                case "DAY":
                     clamping_strategy = DayClampingStrategy()
                     end_date_provider = ClampingEndProvider(
                         DayClampingStrategy(is_ceiling=False),
-                        end_date_provider,
-                        granularity=timedelta(seconds=1),
+                        end_date_provider,  # type: ignore  # Having issues w/ inspection for GapType and CursorValueType as shown in existing tests. Confirmed functionality is working in practice
```

Comment:
Yeah, `CursorType` and `GapType` should probably have been templated of a `Generic` `ConcurrentCursor`. I'm not sure this would have solved the pain to work with MyPy because we still need a way to describe that the relationship between `CursorType` and `GapType`. It's kind of a mess...

Reply:
yeah, we've run into this a few times where whatever type we pass in even though it works, the signature doesn't match. We probably should revisit, but for now i think we'll just have to punt on this